### PR TITLE
Add Tekton Result build configuration for HACBS

### PR DESCRIPTION
**Pull request by avinal**
## Changes
- Build Tekton Results components in 
- Add build PipelineRun push and pull request configurations for api and watcher 
- Pull request to add Tekton Results build config to infra-deployments: redhat-appstudio/infra-deployments#605
- Jira issue: PLNSRVCE-548

Signed-off-by: Avinal Kumar <avinal@redhat.com>

### CODE REVIEW

File: .tekton/tekton-results-watcher-push.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    pipelinesascode.tekton.dev/max-keep-runs: "5"
+spec:
+  params:
+    - name: git-url
+      value: "{{repo_url}}"
+    - name: revision
+      value: "{{revision}}"
+    - name: output-image
+      value: "quay.io/redhat-appstudio/tekton-results-watcher:{{revision}}"
+    - name: path-context
```
Comment (adambkaplan) [inline .tekton/tekton-results-watcher-push.yaml]: What does `path-context` do here? Is this a context directory for a buildah build?

File: .tekton/tekton-results-watcher-pull-request.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    pipelinesascode.tekton.dev/max-keep-runs: "5"
+spec:
+  params:
+    - name: git-url
+      value: "{{repo_url}}"
+    - name: revision
+      value: "{{revision}}"
+    - name: output-image
+      value: "quay.io/redhat-appstudio/pull-request-builds:tekton-results-watcher-{{revision}}"
+    - name: path-context
```
Comment (adambkaplan) [inline .tekton/tekton-results-watcher-pull-request.yaml]: Similar - what does `path-context` do here?

**Review by adambkaplan — CHANGES_REQUESTED** [review]
I would not put our Dockerfiles in `cmd`, since this is a go package. Most developers prefer to keep go packages free of other files (Markdown READMEs perhaps the exception).

I would move this to a separate directory, such as `images`, where Dockerfiles are stored.

Example:

- `images/api/Dockerfile`
- `images/watcher/Dockerfile`

This may impact the `path-context` parameter that you are setting in the HACBS pipeline. I'm not 100% sure on what this does. If this sets the context directory for buildah, then you probably want to omit it because the image build will need all go code from the source root.

File: .tekton/tekton-results-watcher-push.yaml (hunk 1 above)
Comment (avinal) [reply .tekton/tekton-results-watcher-push.yaml]: It seems so, by default it is set to project root. Couldn't find any documentation describing this param but seeing the examples from jvm-build it should point to the base directory for all the code. Code: https://github.com/redhat-appstudio/build-service/blob/ce5cecf8c5e5bad1c66aafae8222341bbe82a22a/controllers/component_build_controller.go#L603, I will omit this.

**Review by adambkaplan — APPROVED** [review]
Looks good! Is there any way we can verify the HACBS build succeeds prior to merge?

**Comment by avinal** [comment]
> Looks good! Is there any way we can verify the HACBS build succeeds prior to merge?

It is kind of a deadlock, configurations are needed in both tektoncd-results and infra-deployments to enable image building. Also, we need to [install the PaC CI App](https://github.com/redhat-appstudio/infra-deployments#component-testing-and-building-of-images) in our repo. I have raised a request, please check.

**Comment by adambkaplan** [comment]
I'll merge this then - if we need to fix anything we can do so in a follow-up PR.

**Pull request merged**